java/src/org/dartlang/vm/service/VmServiceBase.java

```diff
@@ -190,6 +193,18 @@ public static VmService localConnect(int port) throws IOException {
   RequestSink requestSink;
 
   /**
+   * The [addBreakpoint] RPC is used to add a breakpoint at a specific line of some script. This
+   * call works for protocol versions 2.1 and 3.0
```

Comment:
The spec is the "theory" and the VM is the "reality". In theory, theory and reality are the same.... but the Dart SDK 1.12 stable VM is reporting protocol version 2.1.